### ISSUE TITLE
Parallel rendering of React components blocks the document

### ISSUE THREAD

**Issue by sasoria** [withastro/astro#7408]
### What version of `astro` are you using?

2.6.4

### Are you using an SSR adapter? If so, which one?

Node

### What package manager are you using?

pnpm, npm

### What operating system are you using?

Mac

### What browser are you using?

Chrome

### Describe the Bug

When I try to test parallel rendering of React components with promises it seems to block the entire document, until the promises resolve.

``` jsx
await new Promise((res) => setTimeout(res, 1800));

const ReactComponent = () => {
  return (
    <h1>{1800}ms Delayed</h1>
  );
};

export default ReactComponent;
```

Parallel rendring works as intended when I use Astro components. If I'm doing something wrong, please point me in the right direction.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-wsphny?file=src%2Fpages%2Findex.astro

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by bluwy** [comment]
Yeah we currently only support parallel rendering for Astro components, not yet for framework components. cc @johannesspohr

**Comment by sasoria** [comment]
Awesome! Suspense only works client-side, but I want to use it server-side once Astro has support for React server components.

**Comment by natemoo-re** [comment]
I'm going to move this over to our roadmap repo because it's a feature request! This would be great to support for frameworks if possible, but it could be extremely tricky. Unfortunately we need to be able to introspect the output of the framework component in order to support recursive islands, which is why we block for them currently.
